examples/summary.zig

```diff
@@ -0,0 +1,72 @@
+const std = @import("std");
+const zbench = @import("zbench");
+const test_allocator = std.testing.allocator;
+
+fn function1() []const u8 {
+    var result: usize = 0;
+    var i: usize = 0;
+    while (i < 1000) : (i += 1) {
+        const square = i * i;
+        result += square;
+    }
+
+    return "Hello, world!";
+}
+
+fn function2() []const u8 {
+    var result: usize = 0;
+    var i: usize = 0;
+    while (i < 20000) : (i += 1) {
+        const square = i * i;
+        result += square;
+    }
+
+    return "Hello, world!";
+}
+
+fn function3() []const u8 {
+    var result: usize = 0;
+    var i: usize = 0;
+    while (i < 300000) : (i += 1) {
+        const square = i * i;
+        result += square;
+    }
+
+    return "Hello, world!";
+}
+
+fn fn1(_: std.mem.Allocator) void {
+    _ = function1();
+}
+
+fn fn2(_: std.mem.Allocator) void {
+    _ = function2();
+}
+
+fn fn3(_: std.mem.Allocator) void {
+    _ = function3();
```

Comment:
If you wanted, it might help a reader if the `functionX()` function bodies were inlined into `fnX()`, having the extra layer doesn't add anything I think, and just means more bulk to scan. The loop can be shrunk a bit too so there's less to inspect:

```zig
fn fn1(_: std.mem.Allocator) void {
    var result: usize = 0;
    for (0..1000) |i| result += i * i;
}
```

Reply:
Oh yeah, these functions were a little bit useless! Thanks for the suggestion, applied it to the summary tests.

Reply:
Nice, that looks much better as example code now I think :smile: